### DISCUSSION TITLE
Changing data after axios post request?

### DISCUSSION THREAD

**Discussion by impubes** [alpinejs/alpine#3189] [1. Help]
Hello,
I a having strange problem (I am very new to programming) when i trigger the get() function.  Why it is not possible to change the value of null value of dat after getting value from axios post request. It is like considering this.dat like new data and do not want to change the null value of dat ? Maybe some kind of javascript problem (not related to Alpine.js)? 
```

document.addEventListener('alpine:init', () => {Alpine.data
    (
        'func', () =>
    ({
        dat: null,

        get() {
            axios.post('/some/api', {
                firstName: 'Fred',
                lastName: 'Flintstone'
              }).then(function (response) {
               this.dat = response.data;
              }).catch(function (error) {
                console.log(error);
              });
        },
    })
    )
})
```

**Comment by SimoTod** [comment]
This inside your callback is not what you think it should be. Either use an arrow function as a callback or assign this to a variable before calling axios and use that variable in your callback

**Comment by ekwoka** [comment]
It needs to be an arrow function passed into then.

Or use async await (the preferred)
